LDMP/main_widget.py

```diff
@@ -54,10 +62,18 @@ def __init__(self, parent=None):
         super(MainWidget, self).__init__(parent)
 
         self.setupUi(self)
+
         # remove space before dataset item
         self.treeView_datasets.setIndentation(0)
         self.treeView_datasets.verticalScrollBar().setSingleStep(10)
 
+        # instantiate calcluate callback container to reuse most of old code
+        # do this before setting setupAlgorithmsTree
+        self.dlg_calculate_LD = DlgCalculateLD()
+        self.dlg_calculate_TC = DlgCalculateTC()
+        self.dlg_calculate_Biomass = DlgCalculateRestBiomass()
+        self.dlg_calculate_Urban = DlgCalculateUrban()
+
         self.setupAlgorithmsTree()
         self.setupDatasets()
```

Comment:
Populating the list of datasets should be made async since it is a potentially long operation (it will involve scanning the user's hard disk and also contacting the remote GEE server). I realize we are not at that stage yet but nevertheless we can already start laying the ground for it.

Also, perhaps we can defer setting up the existing datasets to when the user actually visits the datasets tab instead of doing this work up front?

Reply:
ok, I'll do when I've the API interface to download algorithms/datasets